config.py

```diff
@@ -8,119 +8,132 @@
-# Enter 0.25 for 25%. If you set MATCH_AREAS to False you can ignore this parameter.
-AREA_MATCHING_THRESHOLD = 0.25
-
-# If you set MATCH_AREAS = True, how far away from the original outlet point should the script look 
-# for a river reach that is a better match in terms of upstream area?
-# Units are decimal degrees (sorry, not a proper distance measurement, this feature could be improved!)
-# 0.1° is about 11 km near the equator, and about 8 km near at a latitude of 45°
-MAX_DIST = 0.075
-
-# Threshold for number of upstream pixels that defines a stream
-# These values worked will in my testing, but you might try changing if the
-# outlet is not getting snapped to a river centerline properly
-THRESHOLD_SINGLE = 500
-THRESHOLD_MULTIPLE = 5000
+from dotenv import load_dotenv
+from pydantic_settings import BaseSettings
+
+# load any configuration which might be available in a .env file
+load_dotenv()
+
+# autoload the settings from the environment. 
+class Config(BaseSettings):
+    # Path to your CSV file with the watershed outlet data
+    OUTLETS_CSV: str = 'outlets_sample.csv'
+
+    # Set to True for "higher resolution" mode or False for "lower resolution."
+    HIGH_RES: bool = True
+
+    # Directory containing the merged, basin-scale MERIT-Hydro flow direction rasters (.tif)
+    # Download from https://mghydro.com/watersheds/rasters
+    # For all paths, do not include a trailing slash.
+    MERIT_FDIR_DIR: str = "data/raster/flowdir_basins"
+
+    # Directory containing the merged, basin-scale MERIT-Hydro flow accumulation rasters (.tif)
+    # Download from https://mghydro.com/watersheds/rasters
+    MERIT_ACCUM_DIR: str = "data/raster/accum_basins"
+
+    # Set to True if you want the script to write status messages to the console
+    VERBOSE: bool = True
+
+    # Set to True to make a bunch of plots of each watershed.
+    # (Just for debugging. Slows down the script a lot.)
+    PLOTS: bool = True
+
+    # Folder where you have stored the Merit-BASINS catchment shapefiles.
+    # These files need to be downloaded from: https://www.reachhydro.org/home/params/merit-basins
+    HIGHRES_CATCHMENTS_DIR: str = "data/shp/merit_catchments"
+
+    # Location of simplified unit catchment boundaries vector data (shapefiles)
+    # Download from: https://mghydro.org/watersheds/share/catchments_simplified.zip
+    LOWRES_CATCHMENTS_DIR: str = "data/shp/catchments_simplified"
+
+    # Folder where you have stored the MERIT-Basins River flowline shapefiles
+    # Download from: https://www.reachhydro.org/home/params/merit-basins
+    RIVERS_DIR: str = "C:/Data/GIS/MERITBasins/rivers"
+
+    # EDIT mmaelicke:
+    # This file was in the original repo and hardcoded in the script.
+    # I move this into the config for reasons of readability and clarity.
+    # This file has the merged "megabasins_gdf" in it
+    MERIT_BASINS_SHP: str = 'data/shp/basins_level2/merit_hydro_vect_level2.shp'
```

Comment:
not this change.

I added this path to the config as it was hard to spot before. On my system the data folder is located somewhere else, thus, I needed to config it.